mlflow/pyfunc/__init__.py

```diff
@@ -345,8 +347,15 @@ def _enforce_schema(pdf: pandas.DataFrame, input_schema: Schema):
     For column types, we make sure the types match schema or can be safely converted to match the
     input schema.
     """
-    if isinstance(pdf, list):
-        pdf = pandas.DataFrame(pdf)
+    if isinstance(pdf, list) or isinstance(pdf, np.ndarray) or isinstance(pdf, dict):
```

Comment:
This line can be simplified to:

```suggestion
    if isinstance(pdf,(list, np.ndarray, dict)):
```